outlook/src/classes/Partner.ts

```diff
@@ -28,25 +37,82 @@ class Partner {
     }
 
     getInitials() : string {
-        if (!this.name) {
-            return "";
-        }
-        const names = this.name.split(" ");
-        let initials = names[0].substring(0, 1).toUpperCase();
-        
-        if (names.length > 1) {
-            initials += names[names.length - 1].substring(0, 1).toUpperCase();
-        }
-        return initials;
+
+        let rgx = new RegExp(/(\p{L}{1})\p{L}+/, 'gu');
+
+        let initials = [...this.name.matchAll(rgx)] || [];
+
+        return (
+            (initials.shift()?.[1] || '') + (initials.pop()?.[1] || '')
+        ).toUpperCase();
+
     };
 
+    /***
+     * Returns True if the partner exists in the Odoo database, False otherwise
+     */
+    isAddedToDatabase(): boolean {
+        return this.id > 0;
+    }
+
+    /***
+     * Creates a partner which is not stored in a Odoo database
+     * @param name
+     * @param email
+     * @param company
+     */
+    static createNewPartnerFromEmail = (name: string, email: string) : Partner => {
+        let partner = new Partner();
+        partner.name = name;
+        partner.email = email;
+        return partner;
+    }
+
     static fromJSON(o: Object): Partner {
         if (!o) return new Partner();
         const partner = Object.assign(new Partner(), o);
         partner.company = Company.fromJSON(o['company']);
         partner.enrichmentInfo = EnrichmentInfo.fromJSON(o['enrichment_info']);
         return partner;
     }
+
+    /***
+     * Given a list of partners having their name and/or email respectively matching a contact's name and/or email,
+     * this method returns a sorted list of the best matched partners, partners are sorted according to these criteria:
+     * 1) partners with an email and name which both equals the provided contact email and name
+     * 2) partners with an email which is equal to the provided contact email
+     * 3) partners with a name which is equal to the provided contact name
+     * @param email the contact's email
+     * @param name the contact's name
+     * @param partners a list of partners
+     */
+    static sortBestMatches(email: string, name: string, partners: Partner[]): Partner[]
```

Comment:
Not really my language so just asking: can't it be simplified a bit with less nested if-else ?

Reply:
Actually yeah, I managed to simplify it using a single if else statement.